service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/CopyControlledAzureDatabaseDefinitionStep.java

```diff
@@ -0,0 +1,182 @@
+import bio.terra.workspace.service.resource.exception.ResourceNotFoundException;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import com.azure.core.management.exception.ManagementException;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CopyControlledAzureDatabaseDefinitionStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(CopyControlledAzureDatabaseDefinitionStep.class);
+
+  private final SamService samService;
+  private final AuthenticatedUserRequest userRequest;
+  private final ControlledAzureDatabaseResource sourceDatabase;
+  private final ControlledResourceService controlledResourceService;
+  private final CloningInstructions resolvedCloningInstructions;
+  private final ResourceDao resourceDao;
+
+  public CopyControlledAzureDatabaseDefinitionStep(
+      SamService samService,
+      AuthenticatedUserRequest userRequest,
+      ControlledAzureDatabaseResource sourceDatabase,
+      ControlledResourceService controlledResourceService,
+      CloningInstructions resolvedCloningInstructions,
+      ResourceDao resourceDao) {
+    this.samService = samService;
+    this.userRequest = userRequest;
+    this.sourceDatabase = sourceDatabase;
+    this.controlledResourceService = controlledResourceService;
+    this.resolvedCloningInstructions = resolvedCloningInstructions;
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    var inputParameters = flightContext.getInputParameters();
+    var workingMap = flightContext.getWorkingMap();
+
+    // get the inputs from the flight context
+    var destinationResourceName =
+        getInputParameterOrWorkingValue(
+            flightContext,
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE_NAME,
+            WorkspaceFlightMapKeys.ResourceKeys.PREVIOUS_RESOURCE_NAME,
+            String.class);
+    var description =
+        getInputParameterOrWorkingValue(
+            flightContext,
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE_DESCRIPTION,
+            WorkspaceFlightMapKeys.ResourceKeys.PREVIOUS_RESOURCE_DESCRIPTION,
+            String.class);
+    var destinationWorkspaceId =
+        getRequired(inputParameters, ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+    var destinationResourceId =
+        getRequired(inputParameters, ControlledResourceKeys.DESTINATION_RESOURCE_ID, UUID.class);
+
+    var destinationManagedIdentity =
+        Optional.ofNullable(
+                resourceDao.getResourceByName(
+                    destinationWorkspaceId, sourceDatabase.getDatabaseOwner()))
```

Comment:
managed identity will have been cloned at this point (see [this PR](https://github.com/DataBiosphere/terra-workspace-manager/pull/1521)). we know that the managed identity will have been cloned because we enforce the order of resource cloning [here](https://github.com/DataBiosphere/terra-workspace-manager/pull/1536/files#diff-456c3140ed43be51fbc37c5b3330eda9926efc43fa49ba287b18e4bcaedd3907R33). 

the source database is owned by a managed identity with the same name, so here we retrieve the managed identity by that name and assign it as the owner of the newly cloned database.